### DISCUSSION TITLE
Error: "No such option '--config-name'"

### DISCUSSION THREAD

**Discussion by apuroopasreeyapureddy-design** [talmolab/sleap#2850] [Help!] [answered]
I have installed Sleap version 1.6.5. Upon trying to train using my labelled files on it, I get this error: 

<img width="1906" height="845" alt="image" src="https://github.com/user-attachments/assets/75895d45-e98e-4d6e-ae84-398e05855e4d" />

Even upgrading sleap (which was suggested as a solution in another case) hasn't helped. The sleap system diagnostics look like shown below. Can you please help me figure out what changes I need to make to be able train and track? Thank you!

SLEAP System Diagnostics
========================
Generated: 2026-08-12 19:50:35

[Platform]
  OS:        Windows 11
  Platform:  Windows-11-10.0.26100-SP0
  Machine:   AMD64
  Processor: Intel64 Family 6 Model 165 Stepping 3, GenuineIntel
  RAM:       12.2 GB / 15.8 GB
  Disk:      184.7 GB / 235.5 GB

[Python]
  Version:     3.13.2
  Executable:  C:\Users\suver lab\AppData\Roaming\uv\tools\sleap\Scripts\python.exe
  Prefix:      C:\Users\suver lab\AppData\Roaming\uv\tools\sleap

[UV]
  Version:         uv 0.12.3 (507230998 2026-08-07 x86_64-pc-windows-msvc)
  Path:            C:\Users\suver lab\.local\bin\uv.EXE
  Cache Dir:       C:\Users\suver lab\AppData\Local\uv\cache
  Tool Dir:        C:\Users\suver lab\AppData\Roaming\uv\tools
  Tool Bin Dir:    C:\Users\suver lab\.local\bin
  Python Dir:      C:\Users\suver lab\AppData\Roaming\uv\python
  Installed Tools: sleap

[UV Config]
  Default Python:    3.13 (C:\Users\suver lab\.python-version)
  Resolved Python:   C:\Users\suver lab\.venv\Scripts\python.exe
  Python Preference: managed (default)
  Resolution:        highest (default)
  Index Strategy:    first-index (default)
  Prerelease:        if-necessary (default)

[GPU / CUDA]
  NVIDIA Driver: Not detected
  PyTorch:       v2.13.0+xpu (CPU)

[Packages]
  sleap:          v1.6.5 (pip)
  sleap-io:       v0.9.2 (pip)
  sleap-nn:       v0.3.3 (pip)
  numpy:          v2.5.2 (pip)
  h5py:           v3.16.0 (pip)
  PySide6:        v6.11.1 (pip)
  opencv-python:  v5.0.0.93 (pip)
  torch:          v2.13.0+xpu (pip)
  imageio:        v2.37.4 (pip)
  imageio-ffmpeg: v0.6.0 (pip)
  av:             v15.1.0 (pip)

[CLI Binaries]
  sleap:
    Path:      C:\Users\suver lab\.local\bin\sleap.EXE
    Real Path: C:\Users\suver lab\.local\bin\sleap.exe
    Source:    pip
  ffmpeg (imageio):
    Path:      C:\Users\suver lab\AppData\Roaming\uv\tools\sleap\Lib\site-packages\imageio_ffmpeg\binaries\ffmpeg-win-x86_64-v7.1.exe
    Source:    imageio-ffmpeg
    Python:    v7.1-essentials_build-www.gyan.dev

[PATH (relevant entries)]
  C:\Users\suver lab\AppData\Roaming\uv\tools\sleap\Lib\site-packages\torch\lib
  C:\Users\suver lab\AppData\Roaming\uv\tools\sleap\Library\bin
  C:\Users\suver lab\AppData\Roaming\uv\tools\sleap\Lib\site-packages\cv2\../../x64/vc17/bin
  C:\Users\suver lab\.local\bin
  C:\Users\suver lab\AppData\Local\Programs\Python\Launcher[/]
  C:\Users\suver lab\AppData\Local\Microsoft\WindowsApps

Copy this output when reporting issues at:
https://github.com/talmolab/sleap/issues

Tip: Use sleap doctor -o to save diagnostics to a file

**Comment by gitttt-1234** [comment]
Hi @apuroopasreeyapureddy-design,

Thanks for sharing the `sleap doctor` output!  Your logs show NVIDIA Driver: Not detected and PyTorch: v2.13.0+xpu. It looks like when SLEAP was installed, it auto-detected an Intel GPU on your machine and installed the Intel "XPU" build of PyTorch instead of a plain CPU build. We currently don't fully support the XPU accelerator (we're working on extending support to Intel GPUs), and I'm wondering if it's silently breaking the pipeline with misleading errors, or if one of the XPU-specific dependencies is causing issues.

To verify, can you try explicitly installing the CPU build?

```
uv tool install --reinstall --python 3.13 "sleap[nn]" --torch-backend cpu
```

Then try starting training again, you can stop it as soon as it passes the first few steps, since training on CPU is painfully slow and we just need to confirm it starts.

Let us know if the error still persists in the cpu build!

Thanks,

Divya

**Reply by apuroopasreeyapureddy-design** [reply]
Hello!

Thanks for the response. I tried this but the same error still persists: 
<img width="924" height="243" alt="image" src="https://github.com/user-attachments/assets/6124ff33-e54f-4283-b24c-497f098bc926" />
<img width="1796" height="791" alt="image" src="https://github.com/user-attachments/assets/914daa50-3f81-4767-bc79-832901f7a386" />
.

What would you suggest I do now?
Thank you!

With regard,
S.R.Apuroopa.

**Reply by gitttt-1234** [reply, marked as answer]
Hi @apuroopasreeyapureddy-design !

I tried to recreate the same issue on my windows system (with the same cpu install command), and it seems to work well on my end. Could you try the below?
- Could you run this and share the output here? 
```
 & "$env:APPDATA\uv\tools\sleap\Scripts\python.exe" -c "import sleap_nn.cli; print('OK: sleap_nn.cli imported successfully')"
```
- Try running `uv cache clean` and `uv tool uninstall sleap`, then reinstall with `uv tool install --python 3.13 "sleap[nn]" --torch-backend cpu`
- If that doesn't work, can you try [installing from source](https://docs.sleap.ai/latest/installation/#developer-setup)?

Let us know if that works!

Thanks,

Divya

**Reply by apuroopasreeyapureddy-design** [reply]
Hello!!

Sorry for the rather delayed response. The training seems to have worked after clearing the cache!!!! I will retry this on another system where I faced the same issue! Also, here is the output for the code lines you mentioned:

PS C:\Users\suver lab>  & "$env:APPDATA\uv\tools\sleap\Scripts\python.exe" -c "import sleap_nn.cli; print('OK: sleap_nn.cli imported successfully')"
OK: sleap_nn.cli imported successfully
PS C:\Users\suver lab> uv cache clean
Clearing cache at: AppData\Local\uv\cache
Removed 112005 files (12.5GiB)
PS C:\Users\suver lab> uv tool uninstall sleap
Uninstalled 8 executables: sleap, sleap-convert, sleap-diagnostic, sleap-inspect, sleap-label, sleap-render, sleap-track, sleap-train
PS C:\Users\suver lab> uv tool install --python 3.13 "sleap[nn]" --torch-backend cpu
warning: The `--torch-backend` option is experimental and may change without warning.
Resolved 225 packages in 4.71s
      Built antlr4-python3-runtime==4.9.3                                                                                                                                                                        Prepared 224 packages in 1m 48s
Installed 225 packages in 18.48s
 + adlfs==2026.8.0
 + aiobotocore==3.9.0
 + aiohappyeyeballs==2.7.1
 + aiohttp==3.14.3
 + aioitertools==0.13.0

Thank you so much!!!!

With regards,
S.R.Apuroopa.
